### PR TITLE
Fix 'download benchmark' workflow

### PR DESCRIPTION
**Pull request by ludfjig**
Currently, PRs are not able to compare performance to main branch, because downloading benchmarks from latest prerelease fails because the name of the artifact is incorrect. See https://github.com/hyperlight-dev/hyperlight/actions/runs/11925609379/job/33237994407#step:16:13 for an example error. 

Here is an example of it now working https://github.com/hyperlight-dev/hyperlight/actions/runs/11928115786/job/33244575940?pr=74#step:16:12. Note that some will still fail, since we haven't published benchmarks with the new naming conventions yet (It will be done as part of prerelease when this PR is merged). 

This pr also renames the matrix.hypervisor such that hyperv is used for windows, and mshv for linux

### CODE REVIEW

**Review by simongdavies — APPROVED** [review]

File: .github/workflows/Benchmarks.yml [hunk 1]
```diff
@@ -34,9 +26,9 @@ jobs:
             os: [self-hosted, Linux, X64, "1ES.Pool=hld-kvm-amd"]
             hypervisor: kvm
             arch: amd
-          - build: linux-hyperv-release-amd
+          - build: linux-mshv-release-amd
```
Comment (danbugs) [inline .github/workflows/Benchmarks.yml:29]: Do we want to name them different things? Afaik, people call it Linux Hyper-V in official documentation (https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/best-practices-for-running-linux-on-hyper-v)?

File: .github/workflows/Benchmarks.yml [hunk 2]
```diff
@@ -17,14 +17,6 @@ jobs:
...
       fail-fast: true
       matrix:
-        build: [
-          windows-2022-release-amd,
-          linux-kvm-release-amd,
-          linux-hyperv-release-amd,
-          windows-2022-release-intel,
-          linux-kvm-release-intel,
-          linux-hyperv-release-intel,
-        ]
```
Comment (danbugs) [inline .github/workflows/Benchmarks.yml:27]: Do you not have to specify the build matrix?

File: .github/workflows/dep_rust.yml [hunk 3]
```diff
@@ -155,15 +151,15 @@ jobs:
...
+        if: runner.os == 'Linux' && matrix.hypervisor == 'mshv'
         run: sudo dnf install gh -y
 
       - name: Install github-cli (Linux ubuntu)
         if: runner.os == 'Linux' && matrix.hypervisor == 'kvm'
         run: sudo apt install gh -y
 
       - name: Download benchmarks from "latest"
-        run: just bench-download ${{ runner.os }} ${{ matrix.hypervisor }} dev-latest # compare to prerelease
+        run: just bench-download ${{ runner.os }} ${{ matrix.hypervisor }} ${{ matrix.arch}} dev-latest # compare to prerelease
```
Comment (danbugs) [inline .github/workflows/dep_rust.yml:162]: Good catch!

File: .github/workflows/Benchmarks.yml (hunk 2 above)
Comment (ludfjig) [reply .github/workflows/Benchmarks.yml:27]: Nope it seems this was redundant

**Pull request merged**